src/spec-configuration/containerFeaturesConfiguration.ts

```diff
@@ -542,21 +521,18 @@ function updateFromOldProperties<T extends { features: (Feature & { extensions?:
 export async function generateFeaturesConfig(params: ContainerFeatureInternalParams, dstFolder: string, config: DevContainerConfig, getLocalFeaturesFolder: (d: string) => string, additionalFeatures: Record<string, string | boolean | Record<string, string | boolean>>) {
 	const { output } = params;
 
+	// Use experimental feature dependencies if explicitly enabled via flag, or if running in CI
+	const useExperimentalFeatureDependencies = params.experimentalFeatureDependencies || !!process.env['CI'];
```

Comment:
My intention here was to run all the CI tests without needing to run the entire set twice. We already pass in this variable when running in ci, but I understand this is a bit hacky.

What do you think would be best?

1) Run _all_ the tests twice, once with the FF and one without.
2) Keep as is (running ci tests just against this new code path).
3) Some other third thing?

Reply:
Another option is to remove the FF altogether.  Functionally this change is supposed to be the same as the existing implementation when Features don't declare a `dependsOn` property.

Reply:
I was thinking about merging the change in behind the experimental flag. When we do that the env variable shouldn't be a common one like 'CI' because that might well be set in other CIs.

We could keep it as-is and remove it before merging to `main`.